ruby/lib/minitest/queue/parse_argv.rb

```diff
@@ -0,0 +1,200 @@
+require 'optparse'
+
+module CI
+  module Queue
+    class ParseArgv
+      Config = Struct.new(
+        :queue_url,
+        :grind_list,
+        :grind_count,
+        :build_id,
+        :namespace,
+        :timeout,
+        :load_paths,
+        :seed,
+        :worker_id,
+        :max_requeues,
+        :max_duration,
+        :requeue_tolerance,
+        :failure_file,
+        :max_consecutive_failures,
+        :failing_test,
+      )
+
+      def self.help
+        parse_argv_instance = CI::Queue::ParseArgv.new({})
+        parse_argv_instance.parser.help
+      end
+
+      def initialize(argv)
+        @argv = argv
+        @config = Config.new
+      end
+
+      def config
+        parser.parse!(@argv)
+        @config
+      end
+
+      def parser
```

Comment:
I think each flag should get it's own method. That should aid readability a little bit.